docs/nx-cloud/reference/launch-templates.md

```diff
@@ -1,42 +1,296 @@
+          echo $PATH 
+          # can invoke cargo directly because it's in the PATH now. 
+          cargo --version
+```
+
+## Private NPM Registry
+
+If your project consumes packages from a private registry, you'll have to set up an authentication step in a custom launch template. We recommend creating an `.npmrc` file. You can pass the auth token from your main agent, so it's available to the agent machines.
+
+```yaml {% fileName=".nx/workflows/agents.yaml" %}
+launch-templates:
+  my-linux-medium-js:
+    resourceClass: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v6'
+    init-steps:
+      - name: Checkout
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/checkout/main.yaml'
+      - name: Auth to Registry
+        script: |
+          echo "@myorg:registry=https://my-private-registry.com/myorg" >> .npmrc
+          echo "//my-private-registry.com/:_authToken=MY_REGISTRY_TOKEN" >> .npmrc
+      - name: Install Node Modules
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/install-node-modules/main.yaml'
+```
+
+Pass `MY_REGISTRY_TOKEN` via `--with-env-vars`
+
+```
+# this assumes MY_REGISTRY_TOKEN is already defined on the main agent
+nx-cloud start-ci-run --distribute-on="5 my-linux-medium-js" --with-env-vars="MY_REGISTRY_TOKEN"
+```
+
+{% callout type="note" title="Token Name" %}
+The token name is not important here, `MY_REGISTRY_TOKEN` is to demonstrate where the value can be placed.
+
+Reference the npm docs on [how to use a `.npmrc` file](https://docs.npmjs.com/cli/v10/configuring-npm/npmrc)
+{% /callout %}
+
+## Custom Node Version
+
+Nx Agents come with node LTS installed. If you want to use a different version, you can create a custom step to install the desired node version. Here is a simplified launch template using [nvm](https://github.com/nvm-sh/nvm) to install a different node version.
+
+```yaml {% fileName=".nx/workflows/agents.yaml" %}
+launch-templates:
+  node-18:
+    resourceClass: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v6'
+    init-steps:
+      - name: Checkout
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/checkout/main.yaml'
+      - name: Install nvm
+        script: |
+          # run nvm install script
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+          # source the updated profile to get the nvm command available
+          source ~/.profile
+          # install the needed version of node via nvm
+          nvm install 21.7.3
+          # echo the current path (which now includes the nvm-provided version of node) to NX_CLOUD_ENV
+          echo "PATH=$PATH" >> $NX_CLOUD_ENV
+      - name: Print node version
+        # confirm that the node version has changed
+        script: node -v
+      - name: Install Node Modules
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/install-node-modules/main.yaml'
+      # Continue setup steps as needed
+```
```

Comment:
oh nice! I think the plan is to eventually bake nvm into the image, and allow users to chose their version via a custom step, similar to [how github actions do it](https://github.com/actions/setup-node?tab=readme-ov-file#usage) 

but this will be perfect to have until then!

Reply:
I can take a stab at it monday.